test/test_python_factory.py

```diff
@@ -0,0 +1,146 @@
+                    "path": f'{python_dir}/factory/',
+                    "targets": {
+                        "1": {
+                            "module": "wsgi",
+                            "callable": "wsgi_a",
+                            "factory": False
+                        },
+                        "2": {
+                            "module": "wsgi",
+                            "callable": "wsgi_b",
+                            "factory": False
+                        },
+                        "factory-1": {
+                            "module": "wsgi",
+                            "callable": "wsgi_a_factory",
+                            "factory": True
+                        },
+                        "factory-2": {
+                            "module": "wsgi",
+                            "callable": "wsgi_b_factory",
+                            "factory": True
+                        },
+                    },
+                }
+            },
+        }
+    )
+
+    resp = client.get(port=8080)
+    assert resp['status'] == 200
+    assert resp['body'] == '1'
+
+    resp = client.get(port=8081)
+    assert resp['status'] == 200
+    assert resp['body'] == '2'
+
+    resp = client.get(port=8082)
+    assert resp['status'] == 200
+    assert resp['body'] == '1'
+
+    resp = client.get(port=8083)
+    assert resp['status'] == 200
+    assert resp['body'] == '2'
+
+def test_python_factory_without_targets():
+    python_dir = f'{option.test_dir}/python'
+
+    assert 'success' in client.conf(
+        {
+            "listeners": {
+                "*:8080": {
+                    "pass": "applications/python-app-factory"
+                },
+                "*:8081": {
+                    "pass": "applications/python-app"
+                }
+            },
+            "applications": {
+                "python-app-factory": {
+                    "type": client.get_application_type(),
+                    "working_directory": f'{python_dir}/factory/',
+                    "path": f'{python_dir}/factory/',
+                    "module": "wsgi",
+                    "callable": "wsgi_a_factory",
+                    "factory": True
+                },
+                "python-app": {
+                    "type": client.get_application_type(),
+                    "working_directory": f'{python_dir}/factory/',
+                    "path": f'{python_dir}/factory/',
+                    "module": "wsgi",
+                    "callable": "wsgi_b",
+                    "factory": False
+                }
+            },
+        }
+    )
+
+    resp = client.get(port=8080)
+    assert resp['status'] == 200
+    assert resp['body'] == '1'
+
+    resp = client.get(port=8081)
+    assert resp['status'] == 200
+    assert resp['body'] == '2'
+
+def test_python_factory_invalid_callable_value(skip_alert):
+    skip_alert(r"[\s\S]*") # This regex will match to any string
```

Comment:
This will skip all alerts, including potentially dangerous ones (like segfaults), so if some alerts are expected, they should be listed explicitly:

```
-    skip_alert(r"[\s\S]*") # This regex will match to any string
+    skip_alert(
+        r"failed to create initial state",
+        r"failed to apply new conf",
+        r"did not return callable object",
+        r"can not be called to fetch callable",
+    )
```

Reply:
Hi @andrey-zelenkov just a question. on my local testing found only these alerts only
r"failed to apply new conf",
r"did not return callable object",
r"can not be called to fetch callable"
so will add skip_alert for them

Did you see it anywhere ("failed to create initial state",) 
while running tests of this file, or you have added it only as an example here.